firefox-ios/WidgetKit/DownloadManager/DownloadLiveActivity.swift

```diff
@@ -143,6 +148,27 @@ struct DownloadLiveActivity: Widget {
         let mbExpected = ByteCountFormatter.string(fromByteCount: bytesExpected, countStyle: .file)
         let subtitle = String(format: .LiveActivity.Downloads.FileProgressText, mbCompleted, mbExpected)
         let totalCompletion = liveDownload.state.completedDownloads == liveDownload.state.downloads.count
+        let circleProgressPercentage = min(
+          liveDownload.state.containsOnlyEncodedFiles ? 1.0 : liveDownload.state.totalProgress, 1.0
+        )
+
+        var downloadTexts: some View {
```

Comment:
Extracted this to a subview to not violate the 45-line closure limit restriction